opm/models/blackoil/blackoilpolymerparams.cpp

```diff
@@ -0,0 +1,403 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+OPM is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+Consult the COPYING file in the top-level source directory of this
+module for the precise wording of the license and the list of
+copyright holders.
+*/
+
+#include <config.h>
+#include <opm/models/blackoil/blackoilpolymerparams.hpp>
+
+#if HAVE_ECL_INPUT
+#include <opm/input/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyadsTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlymaxTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyrockTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyshlogTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyviscTable.hpp>
+#endif
+
+#include <cassert>
+#include <cstddef>
+#include <stdexcept>
+#include <type_traits>
+
+namespace {
+
+#if FLOW_INSTANTIATE_FLOAT && HAVE_ECL_INPUT
```

Comment:
question here, if FLOW_INSTANTIATE_FLOAT is false, then the code will not compile, right?

Reply:
yes, that's the point of the #if's.

Reply:
Somehow locally, I got a config.h file without defining FLOW_INSTANTIATE_FLOAT through IDE cmake.  Let me try again.

Reply:
that's likely because you have not turned on the cmake option to build the float simulators, ie, -DBUILD_FLOW_FLOAT_VARIANTS=1 ?

Reply:
Yes. All the flow variants do not compile without using the cmake flag.

Reply:
ah yes. I see the problem. That was very stupid of me. fix coming up.